### PR TITLE
New packages: RADICAL-Cybertools

### PR DESCRIPTION
**Pull request by mtitov**

### CODE REVIEW

**Comment by adamjstewart** [comment]
Can you split each package into a separate PR? That makes it much easier to review.

**Comment by mtitov** [comment]
hi @adamjstewart, yes, I will (thought that b/c they are all connected to keep them within one PR)

**Comment by adamjstewart** [comment]
Hmm, it's up to you. If they're all connected, it's still easier to review as separate PRs, but harder to merge because things need to happen in order.

P.S. have these maintainers consented to be maintainers?

**Comment by mtitov** [comment]
> P.S. have these maintainers consented to be maintainers?

@adamjstewart all maintainers are original authors 
(for `py-radical-*` maintainer is a core developer from a team I'm in)

File: var/spack/repos/builtin/packages/py-radical-entk/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-entk/package.py]:
```suggestion
    version('develop', branch='devel')
```
Spack has a hard-coded list of versions considered to be newer than stable releases (develop, main, master, trunk).

File: var/spack/repos/builtin/packages/py-radical-entk/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-entk/package.py]: I don't see saga listed as a dep

File: var/spack/repos/builtin/packages/py-radical-entk/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-entk/package.py]:
```suggestion
    depends_on('py-pika@0.13.0',                type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-radical-entk/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-entk/package.py]: Missing dependency on sphinx

File: var/spack/repos/builtin/packages/py-radical-entk/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-entk/package.py]: Is develop required? It seems like any version should work

File: var/spack/repos/builtin/packages/py-radical-gtod/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-gtod/package.py]:
```suggestion
    version('develop', branch='devel')
```

File: var/spack/repos/builtin/packages/py-radical-pilot/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-pilot/package.py]:
```suggestion
    version('develop', branch='devel')
```

File: var/spack/repos/builtin/packages/py-radical-pilot/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-pilot/package.py]:
```suggestion
    depends_on('py-radical-utils@1.6.7:',       type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-radical-pilot/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-pilot/package.py]:
```suggestion
    depends_on('py-radical-saga@1.6.6:',        type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-radical-pilot/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-pilot/package.py]: I don't think develop is actually required is it?

File: var/spack/repos/builtin/packages/py-radical-saga/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-saga/package.py]:
```suggestion
    version('develop',  branch='devel')
```

File: var/spack/repos/builtin/packages/py-radical-saga/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-saga/package.py]: I don't think develop is required

File: var/spack/repos/builtin/packages/py-radical-utils/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-utils/package.py:16]: We don't usually remove older versions without first deprecating them. Is there a reason you need to remove these older versions?

File: var/spack/repos/builtin/packages/py-radical-utils/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-utils/package.py]:
```suggestion
    version('develop', branch='devel')
```

File: var/spack/repos/builtin/packages/py-radical-utils/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-radical-utils/package.py]: develop not required?

File: var/spack/repos/builtin/packages/py-saga-python/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-saga-python/package.py]: The way we normally deprecate packages is to keep the package the same, but change the docstring to note that it is deprecated and add `deprecated=True` to all versions.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-radical-pilot/package.py
Comment (mtitov) [reply var/spack/repos/builtin/packages/py-radical-pilot/package.py]: it is provided in case user wants to set a version with latest changes, thus all `devel` (`develop`) versions will be used instead of releases (developed versions are synced)

File: var/spack/repos/builtin/packages/py-radical-utils/package.py
Comment (mtitov) [reply var/spack/repos/builtin/packages/py-radical-utils/package.py:16]: these versions are completely out of sync and could mislead

File: var/spack/repos/builtin/packages/py-saga-python/package.py
Comment (mtitov) [reply var/spack/repos/builtin/packages/py-saga-python/package.py]: This is the case when package was renamed, and could be kept as alias only (PyPI doesn't have `saga-python` anymore)

File: var/spack/repos/builtin/packages/py-radical-entk/package.py
Comment (mtitov) [reply var/spack/repos/builtin/packages/py-radical-entk/package.py]: right, this was set as an explicit dependency - will remove it (EnTK doesn't make explicit calls for SAGA)

File: var/spack/repos/builtin/packages/py-radical-pilot/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-radical-pilot/package.py]: But users can always get that with `spack install py-radical-pilot@develop ^py-radical-utils@develop ^py-radical-saga@develop`. `depends_on` lists hard requirements, not soft preferences.

File: var/spack/repos/builtin/packages/py-radical-utils/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-radical-utils/package.py:16]: I would still deprecate them before removing, especially since these were previously the default versions used by everyone who installed this package with Spack.

File: var/spack/repos/builtin/packages/py-saga-python/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-saga-python/package.py]: Was `saga-python` renamed to `radical-saga`, or are they different?

File: var/spack/repos/builtin/packages/py-radical-entk/package.py
Comment (mtitov) [reply var/spack/repos/builtin/packages/py-radical-entk/package.py]: `sphinx` shouldn't be there, we'll update setup requirements for entk package accordingly

File: var/spack/repos/builtin/packages/py-radical-pilot/package.py
Comment (mtitov) [reply var/spack/repos/builtin/packages/py-radical-pilot/package.py]: Thank you! yeah, missed that, will remove these dependencies

File: var/spack/repos/builtin/packages/py-radical-utils/package.py
Comment (mtitov) [reply var/spack/repos/builtin/packages/py-radical-utils/package.py:16]: will do, thanks!

File: var/spack/repos/builtin/packages/py-saga-python/package.py
Comment (mtitov) [reply var/spack/repos/builtin/packages/py-saga-python/package.py]: correct `saga-python`-> `radical-saga` and PyPI contains `radical-saga` from version `0.59.0`,
also can confirm we don't have this version (`0.41.3`) within github packages

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-saga-python/package.py]: Hmm, if the package is no longer available on PyPI, and the release can't be downloaded from anywhere else, maybe we should just delete the package. Clearly no one is relying on this if it hasn't existed for some time.

**Comment by mtitov** [comment]
thank you @adamjstewart for feedback! all comments got resolved

**Review by adamjstewart — DISMISSED** [review]
Just waiting for the other new maintainers to review this PR, then I'll merge.

File: var/spack/repos/builtin/packages/py-saga-python/package.py
Comment (andre-merzky) [inline var/spack/repos/builtin/packages/py-saga-python/package.py:15]: @adamjstewart : Should this one be deprecated also instead of plain removing?  I though agree with @mtitov : the version is disfunct and should be removed ASAP.  But from you comment above it sounds like a deprecation period should be observed anyway?

**Review by andre-merzky — DISMISSED** [review]
Added one question about saga-python deprecation, otherwise this looks good to me.  Thanks @mtitov !

File: var/spack/repos/builtin/packages/py-saga-python/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-saga-python/package.py:15]: It's probably best to deprecate it. There may be someone who still has the old tarball in their downloads cache.

**Comment by cf-natali** [comment]
@adamjstewart 

Hi,

I am not familiar with spack - never heard of it before! - so am not really in a position to review this MR.
Feel free to use it, but I don't really intend to be involved in the packaging done here.

Cheers,

File: var/spack/repos/builtin/packages/py-saga-python/package.py
Comment (mtitov) [reply var/spack/repos/builtin/packages/py-saga-python/package.py:15]: ok, rolled it back with some changes (including deprecation status)

**Comment by adamjstewart** [comment]
Thanks @cf-natali 

@mtitov can you remove @cf-natali from the list of maintainers? In the future, please don't add someone as a maintainer without their consent first.

**Comment by mtitov** [comment]
@adamjstewart ok, will update accordingly. I also didn't get a consent from another developer (`pika` developer), should I remove it right away or we can wait for a confirmation?

p.s. apologize for this, I misunderstood the concept of having a maintainer, I thought that the author of a certain package (core developer) should be recognized and be aware of having their package here.

**Comment by adamjstewart** [comment]
> should I remove it right away or we can wait for a confirmation?

Either one is fine. We can always add them as a maintainer later if this gets merged before they respond.

> p.s. apologize for this, I misunderstood the concept of having a maintainer, I thought that the author of a certain package (core developer) should be recognized and be aware of having their package here.

No worries, I realize "maintainer" is an ambiguous term. It's always good to ask the authors/developers if they have any interest in maintaining Spack's build recipe for their package, and serves as good advertising for Spack.

**Comment by mtitov** [comment]
@adamjstewart Thanks! I removed maintainers for both packages (`ntplib`, `pika`), which are dependencies for RCT packages

**Review by adamjstewart — APPROVED** [review]

**Comment by andre-merzky** [comment]
Thanks @adamjstewart for walking us through the process!

**Comment by adamjstewart** [comment]
Thanks for the first-time contribution!

**Pull request merged**
